faunadb-scala/src/main/scala/faunadb/FaunaClient.scala

```diff
@@ -153,7 +154,32 @@ class FaunaClient private (connection: Connection) {
     *         future is returned.
     */
   def query(expr: Expr, timeout: Option[FiniteDuration])(implicit ec: ExecutionContext): Future[Value] =
-    performRequest(json.valueToTree(expr), timeout)
+    query(json.valueToTree(expr), timeout, None, None)
+
+  /**
+   * Issues a query.
+   *
+   * @param expr the query to run, created using the query dsl helpers in [[faunadb.query]].
+   * @param ec the `ExecutionContext` used to run the query asynchronously.
+   * @param timeout the timeout for the current query. It replaces the timeout value set for this
+   *                [[faunadb.FaunaClient]] if any for the scope of this query. The timeout value has
+   *                milliseconds precision.
+   * @param traceId A unique identifier for this query. Adheres to the
+   *                [W3C Trace Context](https://w3c.github.io/trace-context) spec.
+   * @param tags    Key-value pair metadata to associate with this query.
+   * @return A [[scala.concurrent.Future]] containing the query result.
+   *         The result is an instance of [[faunadb.values.Result]],
+   *         which can be cast to a typed value using the
+   *         [[faunadb.values.Field]] API. If the query fails, failed
+   *         future is returned.
+   */
+  def query(
+             expr: Expr,
+             timeout: Option[FiniteDuration],
+             traceId: Option[String],
+             tags: Option[Map[String, String]],
```

Comment:
same comment here about possibly overloading to keep current user code from having to change.

Reply:
Agreed, fixed this as well